File: src/platform-windows/FileCourier.Core/Models/TransferHistoryRecord.cs
namespace FileCourier.Core.Models;

public enum TransferDirection { Sent, Received }
public enum TransferStatus { Completed, Cancelled, Failed }

/// <summary>
/// Persisted record of a completed (or failed/cancelled) transfer, stored in SQLite.
/// </summary>
public class TransferHistoryRecord
{
    public Guid TransferId { get; set; } = Guid.NewGuid();
    public Guid CounterpartyId { get; set; }
    public string CounterpartyName { get; set; } = string.Empty;
    public TransferDirection Direction { get; set; }
    public int TotalFiles { get; set; }
    public long TotalSize { get; set; }
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
    public TransferStatus Status { get; set; }
}

File: src/platform-android/app/src/main/kotlin/com/filecourier/app/data/TransferHistoryRecord.kt
package com.filecourier.app.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "transfer_history")
data class TransferHistoryRecord(
    @PrimaryKey val transferId: String,
    val counterpartyId: String,
    val counterpartyName: String,
    val direction: String, // "Sent" or "Received"
    val itemName: String,
    val itemPath: String,
    val totalFiles: Int,
    val totalSize: Long,
    val timestamp: Long,
    val status: String // "InProgress", "Completed", "Cancelled", "Failed"
)
